vector/src/main/java/im/vector/app/core/services/VectorSyncService.kt

```diff
@@ -137,20 +142,27 @@ class VectorSyncService : SyncService() {
     }
 
     // I do not move or rename this class, since I'm not sure about the side effect regarding the WorkManager
-    class RestartWhenNetworkOn(appContext: Context, workerParams: WorkerParameters) :
-            Worker(appContext, workerParams) {
+    class RestartWhenNetworkOn(
+            appContext: Context,
+            workerParams: WorkerParameters
+    ) : Worker(appContext, workerParams) {
+
         override fun doWork(): Result {
             Timber.d("## Sync: RestartWhenNetworkOn.doWork()")
             val sessionId = inputData.getString(KEY_SESSION_ID) ?: return Result.failure()
             val syncTimeoutSeconds = inputData.getInt(KEY_SYNC_TIMEOUT_SECONDS, BackgroundSyncMode.DEFAULT_SYNC_TIMEOUT_SECONDS)
             val syncDelaySeconds = inputData.getInt(KEY_SYNC_DELAY_SECONDS, BackgroundSyncMode.DEFAULT_SYNC_DELAY_SECONDS)
             val isPeriodic = inputData.getBoolean(KEY_IS_PERIODIC, false)
+
+            // Not sure how to inject a Clock here
+            val clock = DefaultClock()
```

Comment:
we would need to provide this worker via factory like https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/worker/MatrixWorkerFactory.kt#L46

I'm not sure it's worth the effort for just the clock dependency